model/task/db.go

```diff
@@ -985,72 +985,75 @@ func FindByExecutionTasksAndMaxExecution(taskIds []*string, execution int) ([]Ta
 
 type BuildVariantTuple struct {
 	BuildVariant string `bson:"build_variant"`
-	DisplayName  string `bson:"display_name"`
+	DisplayName  string `bson:"build_variant_display_name"`
 }
 
 const VersionLimit = 50
 
-// FindUniqueBuildVariantNamesByTask returns a list of unique build variants names and their display names for a given task name.
-// It attempts to return the most recent display name for each build variant to avoid returning duplicates caused by display names changing.
-// It only checks the last 50 versions that ran for a given task name.
-func FindUniqueBuildVariantNamesByTask(projectId string, taskName string, repoOrderNumber int) ([]*BuildVariantTuple, error) {
-	pipeline := []bson.M{
-		{"$match": bson.M{
-			ProjectKey:     projectId,
-			DisplayNameKey: taskName,
-			RequesterKey:   bson.M{"$in": evergreen.SystemVersionRequesterTypes},
-			"$and": []bson.M{
-				{RevisionOrderNumberKey: bson.M{"$gte": repoOrderNumber - VersionLimit}},
-				{RevisionOrderNumberKey: bson.M{"$lte": repoOrderNumber}},
-			},
+// Reusable pipeline stages when finding build variant display names
+var (
+	// sort build variant display names alphabetically
+	sortByVariantDisplayName = bson.M{
+		"$sort": bson.M{
+			"build_variant_display_name": 1,
 		},
+	}
+
+	// cleanup the results
+	projectBvResults = bson.M{
+		"$project": bson.M{
+			"_id":                        0,
+			"build_variant":              "$" + BuildVariantKey,
+			"build_variant_display_name": "$" + BuildVariantDisplayNameKey,
+		},
+	}
+
+	// reorganize the results to get the build variant names and a corresponding build id
+	projectBuildIdAndVariant = bson.M{
+		"$project": bson.M{
+			"_id":                      0,
+			BuildVariantKey:            bsonutil.GetDottedKeyName("$_id", BuildVariantKey),
+			BuildVariantDisplayNameKey: bsonutil.GetDottedKeyName("$_id", BuildVariantDisplayNameKey),
+			BuildIdKey:                 "$" + BuildIdKey,
 		},
 	}
 
 	// group the build variants by unique build variant names and get a build id for each
-	groupByBuildVariant := bson.M{
+	groupByBuildVariant = bson.M{
```

Comment:
If these are actually only being used in FindUniqueBuildVariantNamesByTask, I'm not sure it's worth pulling out. It's a bit easier to read pipelines when you can just read them sequentially rather than navigating between variables and the function itself.